docs/aws/audit/iamcompliance/rules/root_account_certificates_rotation_remediation.mdx

```diff
@@ -1,6 +1,229 @@
+   response = iam_client.list_server_certificates()
+
+   for cert in response['ServerCertificateMetadataList']:
+       cert_name = cert['ServerCertificateName']
+       cert_upload_date = cert['UploadDate']
+
+       # Check if the certificate is older than the threshold
+       if cert_upload_date < threshold_date:
+           message = f"Certificate {cert_name} is older than {threshold_days} days and should be rotated."
+           print(message)
+           # Send notification
+           sns_client.publish(
+               TopicArn='arn:aws:sns:your-region:your-account-id:your-topic',
+               Message=message,
+               Subject='IAM Certificate Rotation Alert'
+           )
+   ```
+
+By following these steps, you can effectively prevent the misconfiguration of not rotating root account certificates in IAM using Python scripts.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console and open the IAM console at https://console.aws.amazon.com/iam/.
+
+2. In the navigation pane, choose "Users". This will display a list of all IAM users associated with the account.
+
+3. Click on the user name of the root account. This will open the summary page for the root account.
+
+4. In the "Security Credentials" section, check the "Access Keys" list. If there are any access keys that are older than 90 days, it indicates that the root account certificates have not been rotated.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access IAM resources.
+
+2. Once the AWS CLI is set up, you can use the following command to list all the IAM users in your AWS account:
+
+   ```
+   aws iam list-users
+   ```
+
+3. After getting the list of IAM users, you can use the following command to list the signing certificates of each user:
+
+   ```
+   aws iam list-signing-certificates --user-name <username>
+   ```
+
+   Replace `<username>` with the name of the IAM user. This command will return a list of the signing certificates associated with the specified IAM user.
+
+4. To check the age of each certificate, look at the `UploadDate` field in the output of the previous command. If the certificate is older than a certain threshold (for example, 90 days), it should be considered for rotation.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. First, you need to install the AWS SDK for Python (Boto3) if you haven't done so already. You can install it using pip:
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<h2>Secret Detected: Secret Keyword</h2> <a href="https://console.cloudanix.com/code?t=secrets"> Click here for more details</a>